pkgs/by-name/si/sillytavern/package.nix

```diff
@@ -0,0 +1,117 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, makeWrapper
+, pkgs
+}:
+
+let
+wrapperScript = pkgs.writeShellScriptBin "sillytavern-wrapper" ''
+  _name="sillytavern"
+  _BUILDLIBPREFIX="$(dirname "$(readlink -f "$0")")/../lib/sillytavern"
+  _SHAREPREFIX="$(dirname "$(readlink -f "$0")")/../share/sillytavern"
+  _config_dir="$HOME/.config/$_name"
+  _yellow_color_code="\33[2K\r\033[1;33m%s\033[0m\n\33[2K\r\033[1;33m%s\033[0m\n"
+  export NODE_PATH="$_BUILDLIBPREFIX/node_modules"
+
+  fusecleanup()
+  {
+    echo "Cleaning up fusemount"
+    fusermount -qzu "$_config_dir/opt"
+  }
+
+trap fusecleanup 1 2 3 6 15
+
+  fusecleanup
+  mkdir -p "$_config_dir/opt" "$_config_dir/optdiff"
+  echo "Fuse mounting of $_config_dir/optdiff and $_SHAREPREFIX to $_config_dir/opt"
+  ${pkgs.unionfs-fuse}/bin/unionfs -o cow -o umask=000 $_SHAREPREFIX=ro:$_config_dir/optdiff=rw $_config_dir/opt|| exit 1
+
+  printf $_yellow_color_code "If  automatic unmounting fails, run this command:" "fusermount -zu $_config_dir/opt" >&2
+
+  if [ ! -d $_config_dir/optdiff/public/user/images/ ]; then
+    echo "Creating $_config_dir/optdiff/public/user/images/"
+    mkdir -p $_config_dir/optdiff/public/user/images/
+  fi
+
+  if [ ! -d $_config_dir/optdiff/default/ ]; then
+    echo "Creating $_config_dir/optdiff/default/"
+    mkdir -p $_config_dir/optdiff/default/
+  fi
+
+  cp -R $_SHAREPREFIX/public/* $_config_dir/optdiff/public
+  cp -R $_SHAREPREFIX/default/* $_config_dir/optdiff/default
+
+  chmod +rw -R $_config_dir/optdiff
+
+  echo "Entering SillyTavern..."
+  cd "$_config_dir/opt"
+  echo "Starting SillyTavern..."
+
+  node ./server.js
```

Comment:
> Hello @anoadragon453, can you test if this now works?

Here is what I'm currently getting when attempting to run this package:

```
➜ nix shell github:aikooo7/nixpkgs/add-sillytavern#sillytavern

➜ sillytavern                                          
Cleaning up fusemount
fusermount: bad mount point /home/user/.config/sillytavern/opt: No such file or directory
Fuse mounting of /home/user/.config/sillytavern/optdiff and /nix/store/prya4yj7hrhccbhgj7qvawmryqlgr8wy-sillytavern-1.11.4/bin/../share/sillytavern to /home/user/.config/sillytavern/opt
If  automatic unmounting fails, run this command:
fusermount -zu /home/user/.config/sillytavern/opt
Creating /home/user/.config/sillytavern/optdiff/public/user/images/
Creating /home/user/.config/sillytavern/optdiff/default/
Entering SillyTavern...
Starting SillyTavern...
/nix/store/prya4yj7hrhccbhgj7qvawmryqlgr8wy-sillytavern-1.11.4/bin/sillytavern: line 43: node: command not found
Cleaning up fusemount
```

However, upon additionally running `nix shell nixpkgs#nodejs`, running `sillytavern` works as expected!

To have `node` in your wrapper script, I believe you'll need to make `node` available in the user's environment. This might work?

```suggestion
  ${pkgs.nodejs}/bin/node ./server.js
```